mitodl/open-discussions
# Implement moving and removing a widget

#### Pre-Flight checklist


#### What are the relevant tickets?
Part of #1508 

#### What's this PR do?
Implements logic to allow rearranging widgets, removing them, and clicking 'Submit' to finalize the change

#### How should this be manually tested?
As a moderator go to the channel settings icon. Click the icon and you should see a dropdown with a 'Manage widgets' option. Click that option and each widget on the channel should have a trash icon and a drag handle, and there should be cancel and submit buttons on the top of the widget list.

Drag widgets around and confirm that the changes stick. If you click 'Submit' it should refresh the list and keep the changes you made. If you rearrange widgets and click 'Cancel' the widgets should revert to their old locations.

Click the trash icon on a widget then click 'Cancel'. The widget should reappear. If you clicked 'Submit' it should go away permanently.

#### Screenshots (if appropriate)
Desktop:
![widget_move_1](https://user-images.githubusercontent.com/863262/50033548-fa22eb00-ffc6-11e8-823f-44796ec91277.png)
![widget_move_2](https://user-images.githubusercontent.com/863262/50033552-fdb67200-ffc6-11e8-9f5f-5376ac62730b.png)
Mobile is delayed for #1518
